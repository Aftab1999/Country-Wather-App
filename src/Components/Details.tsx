import { Container, Typography, Button } from "@mui/material";
import { Box } from "@mui/system";
import axios from "axios";
// import { AnyRecord } from "dns";
import React, { useState } from "react";
import { useLocation, Link } from "react-router-dom";

const Details = () => {
  let API_KEY = "ac3ce49c3a59e8a93a80d44948c1d528";
  const { state }: any = useLocation();
  const [country, setCountry] = useState<any>("");
  const [weatherData, setWeatherData] = useState<any>("");
  const data = state?.data;
  console.log(data);

  const weatherDetail = async (capital: any) => {
    setCountry(capital);
    const resp = await axios.get(
      `http://api.weatherstack.com/current?access_key=${API_KEY}&query=${capital}`
    );
    setWeatherData(resp.data);
    console.log(resp.data);
  };

  function renderWeatherDetails() {
    return (
      <div>
        <Typography >Weather Information</Typography>
        <div>
          <div>
            <img src={weatherData?.current?.weather_icons[0]} alt="" />
          </div>
          <div className="weather-info">
            Temperature
            <div className="weather-data">
              {weatherData?.current?.temperature}
            </div>
          </div>

          <div className="weather-info">
            Wind Speed
            <div className="weather-data">
              {weatherData?.current?.wind_speed}
            </div>
          </div>
          <div className="weather-info">
            Precip
            <div className="weather-data">{weatherData?.current?.precip}</div>
          </div>
        </div>
        <div>
          <Link style={{ textDecoration: "none",}} id="goback" to="/">
            <Button data-testid="backBtn" sx={{backgroundColor:"lightblue"}} >Back </Button>
          </Link>
        </div>
      </div>
    );
  }

  return (
    <div>
         <Container sx={{ minWidth: 300 }}>
        <Box
         style={{
          backgroundColor: "#e2e4e9", 
          display: "flex",
          flexDirection: "column",
          // alignItems: "center",
          // marginTop: "55px",
          marginLeft: "auto",
          marginRight: "auto",
          // marginRight: "100px",
          height: "550px",
          width:"600px",
          border: "2px solid #a1b4be",
        }}
        >
          <Typography variant="h4" data-testid ='Country_Typo'>Country Details</Typography>
          {data?.map((item: any) => {
            return (
              <div>
                <img
                  src={item?.flags.png}
                  alt={item?.flag}
                  height="140"
                  width="200"
                />
                <div data-testid ='CapiTal' >
                  <strong>Capital :-</strong>
                  {item?.capital}
                </div>
                <div>
                  <strong>Population :-</strong>
                  {item?.population}
                </div>
                <div>
                  <strong> Latlng :-</strong>
                  {item?.capitalInfo.latlng[0]}, {item?.capitalInfo.latlng[1]}
                </div>

                <Button variant="outlined"
                  id="Weather"
                  // title="Weather"
                  data-testid="Weather"
                  onClick={() => weatherDetail(item?.capital)}
                >
                  Capital Weather
                </Button>
                {String(weatherData?.location?.name) ===
                  String(item?.capital) && weatherData
                  ? renderWeatherDetails()
                  : null}
              </div>
            );
          })}
        </Box>
        {/* <button onClick={weatherDetail}>call api</button> */}
      </Container>
    </div>
  );
};
export default Details;
