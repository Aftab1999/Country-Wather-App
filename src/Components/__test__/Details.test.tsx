import React from "react"
import Details from "../Details"
import {render,screen, fireEvent, act} from "@testing-library/react"

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom') as any,
    useNavigate: () => ({ navigate: jest.fn() }),
   useLocation: () => ({  }),
  
  }));

  jest.mock("axios", () => ({
    get: jest.fn((_url, _body) => {
      return new Promise((resolve, reject) => {
        if (
          _url ===
          "http://api.weatherstack.com/current?access_key=ac3ce49c3a59e8a93a80d44948c1d528&query=Delhi"
        ) {
          reject(new Error("Id not found!"));
        } else {
          resolve({
            data: {
                capital: [{ id: "Delhi" }],
              id: "Delhi",
            },
            status: 200,
          });
        }
      });
    }),
  }));

  test("checking snapshot", () => {
    const test1 = render(<Details/>);
    expect(test1).toMatchSnapshot();
  });

  test("checking without crashes", () => {
    const test2 = render(<Details/>);
    expect(test2).toBeTruthy();
  });

it ('checkCapital_WeatherButton', async() =>{
    render(<Details/>)
    expect(screen).toMatchSnapshot();
})

it ('Components exists or not', async() =>{
const root= render(<Details/>)
expect(root).toBeTruthy();
   
})

it("checking for text content", async() => {
const { getByTestId } = render(<Details />)
const text = getByTestId("Country_Typo")
expect(text.textContent).toBe("Country Details")

})

// test("checking Capital Weather button", async () => {
//     const { getByTestId } = render(<Details/>);
//     const randomSubmit = getByTestId("Weather");
//     await act(async () => {
//       await fireEvent.click(randomSubmit);
//     });
//     expect(randomSubmit).toBeDefined();
//   });


// it("checking for text content", async() => {
// const { getByTestId } = render(<Details />)
// const text = getByTestId("Capial")
// expect(text.textContent).toBe("Capital")

// })

// it("checking for text content", async() => {
// const { getByTestId } = render(<Details />)
// const text = getByTestId("Weather_Information")
// expect(text.textContent).toBe("Weather Information")

// })




// it ("Check  Capital_Weather", ()=>{
//     const {getByText} = render(<Details/>);
//     const btn:any =getByText('Capital Weather');
//     expect(btn).toBeTruthy();
//     expect (getByText('Capital Weather')).toBeInTheDocument()
// })
// it ("Check Back Button", ()=>{
//     const {queryByTitle} = render(<Details/>);
//     const btn:any = queryByTitle("Back");
//     expect(btn).toBeTruthy();
// }) 
