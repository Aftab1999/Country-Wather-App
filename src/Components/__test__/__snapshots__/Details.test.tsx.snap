// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkCapital_WeatherButton 1`] = `
Object {
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "logTestingPlaygroundURL": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`checking snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-dhgzs6-MuiContainer-root"
        >
          <div
            class="MuiBox-root css-0"
            style="background-color: rgb(226, 228, 233); display: flex; flex-direction: column; margin-left: auto; margin-right: auto; height: 550px; width: 600px; border: 2px solid #a1b4be;"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
              data-testid="Country_Typo"
            >
              Country Details
            </h4>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-dhgzs6-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-0"
          style="background-color: rgb(226, 228, 233); display: flex; flex-direction: column; margin-left: auto; margin-right: auto; height: 550px; width: 600px; border: 2px solid #a1b4be;"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
            data-testid="Country_Typo"
          >
            Country Details
          </h4>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
